src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -110,134 +116,96 @@ public ArmSubsystem() {
-                CANSparkMax.ControlType.kPosition,
-                elbow_pid.slot);
+        automaticPIDAdjuster();
 
-        m_shoulder.getPIDController().setReference(position.shoulder,
-                CANSparkMax.ControlType.kPosition,
-                shoulder_pid.slot);
-
-        // Update the dashboard
-        SmartDashboard.putNumber("Shoulder Position", m_shoulder.getEncoder().getPosition());
-        SmartDashboard.putNumber("Elbow Position", m_elbow.getEncoder().getPosition());
-
-        handle.getEntry("/ArmSubsystem/DesiredState").setString(String.valueOf(m_state_desired));
-        handle.getEntry("/ArmSubsystem/ActualState").setString(String.valueOf(m_state_actual));
-    }
-
-    private static boolean inRange(double value, double target, double tolerance) {
-        return value >= target - tolerance && value <= target + tolerance;
+        m_elbow.getPIDController().setReference(ArmConstants.ELBOW_GEAR_RATIO(m_desiredState.elbow),
+                ControlType.kPosition);
+        m_shoulder.getPIDController().setReference(
+                ArmConstants.SHOULDER_GEAR_RATIO(m_desiredState.shoulder), ControlType.kPosition);
     }
 
-    private void matchActualState() {
-        // Record the actual_state to the state that the arm is in based on encoder values
-        double shoulder_pos = m_shoulder.getEncoder().getPosition();
-        double elbow_pos = m_elbow.getEncoder().getPosition();
-
-        if (inRange(shoulder_pos, REST.shoulder, stateTolerance) && inRange(elbow_pos, REST.elbow, stateTolerance))
-            m_state_actual = State.Rest;
-        else if (inRange(shoulder_pos, PICKUP.shoulder, stateTolerance) && inRange(elbow_pos, PICKUP.elbow, stateTolerance))
-            m_state_actual = State.Pickup;
-        else if (inRange(shoulder_pos, PLACING_A.shoulder, stateTolerance) && inRange(elbow_pos, PLACING_A.elbow, stateTolerance))
-            m_state_actual = State.PlacingA;
-        else if (inRange(shoulder_pos, PLACING_B.shoulder, stateTolerance) && inRange(elbow_pos, PLACING_B.elbow, stateTolerance))
-            m_state_actual = State.PlacingB;
-        else if (inRange(shoulder_pos, PLACING_C.shoulder, stateTolerance) && inRange(elbow_pos, PLACING_C.elbow, stateTolerance))
-            m_state_actual = State.PlacingC;
-        else if (inRange(shoulder_pos, PLACING_D.shoulder, stateTolerance) && inRange(elbow_pos, PLACING_D.elbow, stateTolerance))
-            m_state_actual = State.PlacingD;
-        else if (inRange(shoulder_pos, PLACING_E.shoulder, stateTolerance) && inRange(elbow_pos, PLACING_E.elbow, stateTolerance))
-            m_state_actual = State.PlacingE;
-
-        if (m_state_actual != m_state_desired)
-            m_state_actual = State.Transit;
+    /*
+     * A method that automatically adjusts the slot of the PID controller based on the desired
+     * state. To be called periodically.
+     */
+    private void automaticPIDAdjuster() {
+        // TODO
     }

```

Comment:
I don't know what the logic will be behind this method.

Reply:
Ping me when this is implemented this is the important part.

Reply:
I can't implement it if I don't know the logic behind the switching :).

Reply:
The logic is that we are fundamentally dealing with two different issues when the arm is moving to a position and staying in a position. It may not be the "right" solution but by tuning two different PID's one designed for transitioning between states and another for keeping precision at our present state then we can get the best for both. So when it's at state we configure the motor controller to operate with a more aggressive PID than moving between states. This has been tested and works quite well at keeping a stable position resistant to outside forces.

All of the initial values should change slightly but the logic should remain exactly the same as the first revision of the subsystem.

Does this explain the logic well enough?

Reply:
So when it’s at state use slot 1, else use slot 0? That’s all I needed to know lol.